lib/bundler/dsl.rb

```diff
@@ -106,13 +106,29 @@ def gem(name, *args)
         if current.requirement != dep.requirement
           unless deleted_dep
             return if dep.type == :development
+
+            update_prompt = ""
+            change_version = ". You may also need to change the version requirement specified in the Gemfile if it's too restrictive."
```

Comment:
this is only used once, so it can be inlined into that location, instead of sitting at the outer scope

Reply:
When [this](https://github.com/agrim123/bundler/blob/9a7328ba1275eb60b36212b91bee8e8700f07ec7/lib/bundler/dsl.rb#L113) condition is false then [this](https://github.com/agrim123/bundler/blob/9a7328ba1275eb60b36212b91bee8e8700f07ec7/lib/bundler/dsl.rb#L123-L125) would throw error.

Reply:
that exception message doesn't include `change_version` ?

Reply:
@agrim123 Let's move line 111 into line 119 where it's used. That way ruby doesn't allocate memory for that string when it's not needed. This is what @segiddins is getting at.

Reply:
Oh! Sorry. Fixing it